libr/include/r_types.h

```diff
@@ -305,6 +309,20 @@ typedef void (*PrintfCallback)(const char *str, ...);
 
 #define R_FREE(x) { free(x); x = NULL; }
 
+/* ripped from https://stackoverflow.com/questions/20991229/how-to-freeing-pointers-using-macro-in-c */
+#define R_FREE_ALL(...) \
+do { \
+    int i=0;\
+    void *pta[] = {__VA_ARGS__}; \
+    for(i=0; i < sizeof(pta)/sizeof(void*); i++) \
+    { \
+        if (pta[i])\
+        free(pta[i]); \
```

Comment:
there is not need to check against NULL, free already does. USE R_FREE instead of free + set null